### DISCUSSION TITLE
Does paketo-buildpacks only support ubuntu?

### DISCUSSION THREAD

**Discussion by doddisam** [paketo-buildpacks/feedback#323] [Stacks Team]
Hi Team,

Just wanted to clarify the following:

Does the Paketo Buildpack and Stack only support Ubuntu OS?
If not, does it support any other OS, such as Debian?

**Comment by dmikusa** [comment]
The stacks that we publish are all Ubuntu-based, the project does not publish any other stacks.

That said, many of our buildpacks will run on other versions of Linux. The buildpack binaries themselves are Go static binaries, so they should run on basically any Linux. That means, where a buildpack runs really boil down to the dependencies that the buildpacks is installing & where those run.

Take Java for example, most JVM vendors publish versions of Java that can run on a pretty wide variety of Linux distributions. Thus, if you want to build your Java app and provide your own stack based on a different Linux distro like Debian or Redhat UBI, you can do that and there's a very strong chance that things will just work. For different languages, you'll just need to check the buildpack and the binaries they use for compatibility.

The other option, which is specific to Java, Node.js & UBI, is to try this community project https://github.com/paketo-community/builder-ubi-buildpackless-base. That is specific to UBI though, so if you're looking for Debian that wouldn't be a solution.

Hope that helps!

**Comment by doddisam** [comment]
Hi Daniel,

Thanks for the reply.

Any reason why project chose Ubuntu not Debian?

Cheers!!
sameer

**Reply by dmikusa** [reply]
The decision predates me, but I suspect it's historical. Cloud Foundry had rootfs' based on Ubuntu, and Paketo shares a lot of lineage from Cloud Foundry buildpacks.
